src/main/java/aquality/selenium/core/elements/ElementFinder.java

```diff
@@ -0,0 +1,77 @@
+package aquality.selenium.core.elements;
+
+import aquality.selenium.core.elements.interfaces.IElementFinder;
+import aquality.selenium.core.localization.ILocalizedLogger;
+import aquality.selenium.core.waitings.IConditionalWait;
+import com.google.inject.Inject;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.TimeoutException;
+import org.openqa.selenium.WebElement;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Implementation of IElementFinder.
+ */
+public class ElementFinder implements IElementFinder {
+    private final ILocalizedLogger localizedLogger;
+    private final IConditionalWait conditionalWait;
+
+    @Inject
+    public ElementFinder(ILocalizedLogger localizedLogger, IConditionalWait conditionalWait) {
+        this.localizedLogger = localizedLogger;
+        this.conditionalWait = conditionalWait;
+    }
+
+    @Override
+    public List<WebElement> findElements(By locator, DesiredState desiredState, Long timeoutInSeconds) {
+        AtomicBoolean wasAnyElementFound = new AtomicBoolean(false);
```

Comment:
why do we need this variable? why can't we operate only with 'resultElements.isEmpty()'?
or it is just because lambda usage?

Reply:
result elements is a different array, in fact it has elements after filtering, while this variable defines whether any element was there before filtering. please recheck your suggestion